src/gui/qgspropertyoverridebutton.cpp

```diff
@@ -725,22 +733,46 @@ void QgsPropertyOverrideButton::enableEnabledWidgets( bool enable )
   }
 }
 
-void QgsPropertyOverrideButton::visibleVisibleWidgets( bool visible )
+void QgsPropertyOverrideButton::updateVisibleWidgets( bool visible )
 {
   Q_FOREACH ( const TiedWidget &tw, mVisibleWidgets )
   {
     tw.mWidgetPointer.data()->setVisible( tw.mNatural ? visible : !visible );
   }
 }
 
-void QgsPropertyOverrideButton::setActive( bool active )
+void QgsPropertyOverrideButton::updateExpressionWidgets()
+{
+  Q_FOREACH ( const QPointer<QWidget> &wp, mExpressionWidgets )
+  {
+    QLineEdit *le = qobject_cast<QLineEdit *>( wp.data() );
+    if ( le )
+    {
+      le->setText( mProperty.asExpression() );
+      continue;
+    }
+    QTextEdit *te = qobject_cast<QTextEdit *>( wp.data() );
+    if ( te )
+    {
+      te->setText( mProperty.asExpression() );
+      continue;
+    }
+  }
+}
+
+void QgsPropertyOverrideButton::setActive( bool active, bool forceSignal )
```

Comment:
I'm not a huge fan of these "forceSignal" type arguments - they tend to quickly turn into messy corner cases. 

Here's an alternative idea - why not require that that init is called before registering widgets? Then the various widget registration functions can perform the disabling/checking/etc immediately upon registration.